azurerm/internal/services/storage/resource_arm_hpc_cache_nfs_target.go

```diff
@@ -0,0 +1,288 @@
+package storage
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+
+	"github.com/Azure/azure-sdk-for-go/services/storagecache/mgmt/2019-11-01/storagecache"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/storage/parsers"
+	storageValidate "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/storage/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func namespaceJunctionResource() *schema.Resource {
```

Comment:
I'm not sure why this is a separate function? could we just inline it with the rest of the schema?

Reply:
This function is also used to feed to `schema.HashResource()`, which in turn passed to `schema.NewSet()` as the 1st argument, in this way we have no need to write a dedicated hash function.

Reply:
As it is already handled by `d.Set()`, we can just inline it to schema.